packages/algorand-js/src/interpreter/interpreter.ts

```diff
@@ -1,91 +1,36 @@
-import { mkTransaction } from "algob";
-import type { ExecParams } from "algob/src/types";
-import { AccountState, assignGroupID, SSCParams } from "algosdk";
 import { assert } from "chai";
 
-import { mockSuggestedParams } from "../../test/mocks/txn";
 import { TealError } from "../errors/errors";
 import { ERRORS } from "../errors/errors-list";
+import { Runtime } from "../index";
 import { DEFAULT_STACK_ELEM } from "../lib/constants";
 import { Stack } from "../lib/stack";
-import type { Operator, StackElem, TEALStack, Txn } from "../types";
+import { parser } from "../parser/parser";
+import type { Operator, StackElem, State, TEALStack } from "../types";
+import { Context } from "../types";
 import { BIGINT0, Label } from "./opcode-list";
 
 export class Interpreter {
   readonly stack: TEALStack;
   bytecblock: Uint8Array[];
   intcblock: BigInt[];
   scratch: StackElem[];
-  tx: Txn;
-  gtxs: Txn[];
-  accounts: Map<string, AccountState>;
-  globalApps: Map<number, SSCParams>;
   instructions: Operator[];
   instructionIndex: number;
   args: Uint8Array[];
+  ctx: Context; // interpreter's 'local' context
```

Comment:
yes, but it still needs to assigned as an interpreter's property, right? Otherwise how will the opcodes access the data in `ctx`?
(One option is to pass `ctx` to each `execute()` of opcode)

Reply:
If we are passing context to execute, i don't think we need to assign interpreter's property because `ctx` itself has all the data.

Reply:
also if we are passing `runtime` object to `interpreter`, we can access `ctx` using `runtime.ctx`

Reply:
@amityadav0 the idea here is that interperter will use a 'copy' of store. Interpreter shouldn't use `runtime.store` directly. `ctx` is kept inside interpreter. This is because if the transaction fails, then we ignore the changes in the interpreter's local ctx, and the store remains unchanged.

Reply:
> also if we are passing runtime object to interpreter, we can access ctx using runtime.ctx